src/metabase/lib/equality.cljc

```diff
@@ -136,13 +136,17 @@
       (clojure.core/= (column-join-alias column) join-alias)))
 
 (mu/defn- plausible-matches-for-name :- [:sequential ::lib.schema.metadata/column]
-  [[_ref-kind _opts ref-name :as a-ref] :- ::lib.schema.ref/ref
+  [[_ref-kind opts ref-name :as a-ref] :- ::lib.schema.ref/ref
    columns                              :- [:sequential ::lib.schema.metadata/column]]
   (or (not-empty (filter #(and (clojure.core/= (:lib/desired-column-alias %) ref-name)
                                (matching-join? a-ref %))
                          columns))
       (filter #(and (clojure.core/= (:name %) ref-name)
-                    (matching-join? a-ref %))
+                    ;; TODO: If the target ref has no join-alias, AND the source is fields or card, the join
```

Comment:
This is the same comment as below

Reply:
Because this is the default behavior now there are hundreds of tests for this :)